script.js

```diff
@@ -1,4 +1,159 @@
+/* Two-player game
+Player 1 will press submit and 2 dice are rolled
+Player 1 to select dice order
+Player 2 will press submit and 2 dice are rolled
+Player 2 to select dice order
+To determine who the winner is */
+
+var playerOneChoice = "";
+var playerOneDiceRoll = "";
+var playerOneDiceOne = "";
+var playerOneDiceTwo = "";
+
+var playerTwoChoice = "";
+var playerTwoDiceRoll = "";
+var playerTwoDiceOne = "";
+var playerTwoDiceTwo = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // refresh all variables to play again
+  if (playerTwoChoice !== "") {
+    playerOneChoice = "";
+    playerOneDiceRoll = "";
+    playerOneDiceOne = "";
+    playerOneDiceTwo = "";
+
+    playerTwoChoice = "";
+    playerTwoDiceRoll = "";
+    playerTwoDiceOne = "";
+    playerTwoDiceTwo = "";
+  }
+
+  // player one has not rolled the dice
+  if (playerOneDiceRoll == "") {
+    playerOneDiceOne = rollDices();
+    playerOneDiceTwo = rollDices();
+    playerOneDiceRoll = "rolled";
+    return `Welcome Player 1! <br><br>
+    You rolled...<br>
+    # Dice One: ${playerOneDiceOne}<br>
+    # Dice Two: ${playerOneDiceTwo} <br><br>
+    Please type in either '1' or '2' to continue... <br>
+    # Option 1: ${playerOneDiceOne}${playerOneDiceTwo} <br>
+    # Option 2: ${playerOneDiceTwo}${playerOneDiceOne}`;
+  }
+
+  // if player one has rolled the dice, ask player one to choose arrangement
+
+  // if player one chooses option 1 (arrangement of dice 1+2)
+  if (playerOneDiceRoll == "rolled" && playerOneChoice == "" && input == "1") {
+    playerOneChoice = Number(
+      String(playerOneDiceOne) + String(playerOneDiceTwo)
+    );
+    return `Player 1, your choice is ${playerOneChoice}. <br><br>
+      Player 2, please roll the dices.`;
+  }
+
+  // if player one chooses option 2 (arrangement of dice 2+1)
+  if (playerOneDiceRoll == "rolled" && playerOneChoice == "" && input == "2") {
+    playerOneChoice = Number(
+      String(playerOneDiceTwo) + String(playerOneDiceOne)
+    );
+    return `Player 1, your choice is ${playerOneChoice}. <br><br>
+      Player 2, please roll the dices.`;
+  }
+
+  // if player one does not choose any of the options
+  if (playerOneChoice == "") {
+    return `Player 1, please type in either '1' or '2' to continue... <br><br>
+      # Option 1: ${playerOneDiceOne}${playerOneDiceTwo} <br>
+      # Option 2: ${playerOneDiceTwo}${playerOneDiceOne}`;
+  }
```

Comment:
this is fine but a better way to organise it in terms of pseudo code:

```
if (playerOneDiceRoll == 'rolled') {
   if (input is not 1 and is not 2) {
      return <input validation msg>
    }
    
    playerOneChoice = Number(
      String(playerOneDiceOne) + String(playerOneDiceTwo)
    );
    
    if (input == 2) {
       playerOneChoice = Number(
          String(playerOneDiceTwo) + String(playerOneDiceOne)
        );
     }
     
     return <msg>
 ```

Reply:
things to note:
- input validation usually at the top
- benefit of putting it at the top and returning early - we know that the input is def 1 or 2 now, so we do not need to check for both cases explicitly. we can just assume and assign the value to playerOneChoice as if input was 1, then do a check if its 2 and reassign as necessary.
- one overall return of your msg at the bottom after assigning the appropriate value to playerOneChoice